### PR TITLE
Removing unnecessary lambdas part 39

### PR DESCRIPTION
**Pull request by Piotrek321**

### CODE REVIEW

File: src/blockchain/node/UpdateTransaction.cpp [hunk 1]
```diff
@@ -123,11 +116,7 @@ auto UpdateTransaction::disconnected() const noexcept
 auto UpdateTransaction::EffectiveBestBlock(const block::Height height) const
     noexcept(false) -> block::Hash
 {
-    try {
-
-        return best_.at(height);
-    } catch (...) {
-    }
+    if (best_.count(height)) return best_.at(height);
```
Comment (wiesiekpap) [inline src/blockchain/node/UpdateTransaction.cpp]: Don't know if it is possible with that robin hood map, but generally, for the sake of efficiency, one would prefer
if(auto i = best_.find(height); i != best_.end()) return i->second;

File: src/blockchain/p2p/peer/Address.hpp [hunk 2]
```diff
@@ -65,9 +65,9 @@ class Address
 
     Address(pointer address) noexcept;
 
-    virtual ~Address();
+    virtual ~Address() = default;
```
Comment (wiesiekpap) [inline src/blockchain/p2p/peer/Address.hpp]: This does make a bit of difference, though: now you cannot put a breakpoint on the destructor, which I actually did some time ago chasing a shutdown crash.

**Review by wiesiekpap — APPROVED** [review]
There is a minor efficiency issue, I leave it to you if you want to improve it.

File: src/blockchain/node/UpdateTransaction.cpp (hunk 1 above)
Comment (Piotrek321) [reply src/blockchain/node/UpdateTransaction.cpp]: definitelly. I'll change it in other places where I used this pattern too

**Pull request merged**